backend/core/src/shared/units-transformations.ts

```diff
@@ -115,107 +115,109 @@ const hmiData = (units: Units, byUnitType: UnitSummary[], amiPercentages: string
-        )
-        const rentRange = minMaxValue(
-          {
-            min: parseFloat(summary.rentRange?.min),
-            max: parseFloat(summary.rentRange?.max),
-          },
-          Number.parseFloat(unit.monthlyRent)
-        )
-        summary.rentRange = {
-          min: rentRange.min.toPrecision(2),
-          max: rentRange.max.toPrecision(2),
-        }
-        if (unit.floor) {
-          summary.floorRange = minMaxValue(summary.floorRange, unit.floor)
-        }
-        summary.areaRange = minMaxValue(summary.areaRange, parseFloat(unit.sqFeet))
-        if (unit.status == "available") {
-          summary.totalAvailable += 1
-        }
 
-        return summary
-      }, summary)
+  summary.occupancyRange = minMaxValue(summary.occupancyRange, unit.minOccupancy, unit.maxOccupancy)
+  summary.rentAsPercentIncomeRange = minMaxValue(
+    summary.rentAsPercentIncomeRange,
+    parseFloat(unit.monthlyRentAsPercentOfIncome)
+  )
+  const rentRange = minMaxValue(
+    {
+      min: parseFloat(summary.rentRange?.min),
+      max: parseFloat(summary.rentRange?.max),
+    },
+    Number.parseFloat(unit.monthlyRent)
+  )
+  summary.rentRange = {
+    min: rentRange.min.toPrecision(2),
+    max: rentRange.max.toPrecision(2),
+  }
+  if (unit.floor) {
+    summary.floorRange = minMaxValue(summary.floorRange, unit.floor)
+  }
+  summary.areaRange = minMaxValue(summary.areaRange, parseFloat(unit.sqFeet))
 
-      if (finalSummary.minIncomeRange) {
-        finalSummary.minIncomeRange = minMaxInCurrency(finalSummary.minIncomeRange)
-      }
-      if (finalSummary.rentRange) {
-        finalSummary.rentRange = minMaxInCurrency(finalSummary.rentRange)
-      }
+  if (summary.minIncomeRange) {
+    summary.minIncomeRange = minMaxInCurrency(summary.minIncomeRange)
+  }
+  if (summary.rentRange) {
+    summary.rentRange = minMaxInCurrency(summary.rentRange)
+  }
+  return summary
+}
 
-      return finalSummary
+const summarizeUnits = (units: Units, reservedType?: string): UnitSummary[] => {
+  if (!reservedType) {
+    reservedType = null
+  }
+  const summaries: UnitSummary[] = []
+  let currentUnitType = units[0].unitType
+  let currentUnitRent = units[0].monthlyRent
+  let numAvailable = 0
+
+  // Assumes units are listed in groups by unitType and rent type
+  // Allows for multiples rows under one unit type if the rent methods differ
+  units.forEach((unit, index) => {
+    if (unit.unitType !== currentUnitType || unit.monthlyRent !== currentUnitRent) {
+      const summary = getUnitSummary(units[index - 1])
+      summary.totalAvailable = numAvailable
+      summaries.push(summary)
+      currentUnitType = unit.unitType
+      currentUnitRent = unit.monthlyRent
+      numAvailable = 1
+    } else {
+      numAvailable += 1
     }
-  )
+    if (index === units.length - 1) {
+      const summary = getUnitSummary(unit)
+      summary.totalAvailable = numAvailable
+      summaries.push(summary)
+      return
+    }
+  })
```

Comment:
Can we rely on units being grouped in this way? I understand that's probably the case while we're manually uploading listings, but I don't think this is a long term solution.